www/delivery_dev/ck.php

```diff
@@ -76,11 +79,37 @@
         $creativeId[$i] = 0;
     }
     if (($adId[$i] > 0 || $adId[$i] == -1) && ($conf['logging']['adClicks']) && !(isset($_GET['log']) && ($_GET['log'] == 'no'))) {
+        // If no-click/redirect on inactive banners is enabled, check to see
+        // if the banner is active. If not, exit click processing at this point,
+        // without recording the click or performing redirection
+        if (isset($GLOBALS['conf']['logging']['blockInactiveBannerClicks'])) {
+            $exitProcessing = true;
+            $aZoneLinkedAds = MAX_cacheGetZoneLinkedAds($zoneId);
```

Comment:
Hmm, I don't think a GetZoneLinkedAds should be performed here: the ad or campaing could have been recently unlinked and a perfectly valid click discarded because of it. As I see it, the best alternative would be to add something like `m.status AS campaign_status` to `OA_Dal_Delivery_getAd` and use `MAX_cacheGetAd`, plus additioan checks on the two status fields.